### PR TITLE
Setup database

### PR DESCRIPTION
**Pull request by sharminelow**
Storage class and basic user model defined.

### CODE REVIEW

File: app_server/app/models/Config.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+ * Edit to your local mysql database's credentials
+ */
+
+var config = {};
+
+config.name = '';
+config.username = '';
+config.password = '';
+config.host = 'localhost';
+config.dialect = 'mysql';
```
Comment (yj-soh) [inline app_server/app/models/Config.js]: These may be better done via some sort of template file that's not under version control, or environment variables.

File: app_server/app/models/Storage.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  ];
+
+  // import each model
+  models.forEach(function(model) {
+    logger.info(model + " imported");
+    db[model] = sequelize.import(__dirname + '/' + model);
+  });
+
+  Object.keys(db).forEach(function(modelName) {
+    if ("associate" in db[modelName]) {
```
Comment (yj-soh) [inline app_server/app/models/Storage.js]: Is `models` equivalent to `Object.keys(db)`?
If it is, how about chaining it to the `forEach()` above?
Using the same variable name in the function parameter would be good in that case too.

Also, single quotes for strings.

File: app_server/app/models/Storage.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  var Sequelize = require('sequelize');
+  var config = require('./Config');
+  var db = {};
+
+  // initialize database connection
+  var sequelize = new Sequelize(
+    config.name,
+    config.username,
+    config.password, {
+      host: config.host,
```
Comment (yj-soh) [inline app_server/app/models/Storage.js]: `(` should be 4 spaces as it's a continuation. `{` and `[` are 2.

File: app_server/app/models/Config.js (hunk 1 above)
Comment (sharminelow) [inline app_server/app/models/Config.js]: Shifted this config template file over to app_server/config. I will add it into git ignore after everyone has a copy of this.

File: app_server/app/models/Storage.js (hunk 2 above)
Comment (sharminelow) [inline app_server/app/models/Storage.js]: They differ. models is referred to an array of strings, which are the models' respective names while the keys of db are the model objects after importing.

Have altered the double quotes, thanks for spotting that.

File: app_server/app/models/Storage.js (hunk 3 above)
Comment (sharminelow) [inline app_server/app/models/Storage.js]: Noted and changed

**Comment by sharminelow** [comment]
Can merge?

File: app_server/app/models/Storage.js (hunk 2 above)
Comment (yj-soh) [inline app_server/app/models/Storage.js]: Er, upon another look, I'm not quite sure what you're doing here.
`db` is an object with its keys connected to each model, right?
So you import models into `db` according to the contents of `models`, linking each of them as keys in `db` connected to a [Model](http://docs.sequelizejs.com/en/latest/api/model/)?
The Model class doesn't have any `associate` property in it, so I'm not quite sure what you're doing in the second forEach loop.

**Comment by yj-soh** [comment]
Er, upon another look, I'm not quite sure what you're doing in that second forEach loop in Storage.js.

`db` is an object with its keys connected to each model, right?
So you import models into `db` according to the contents of `models`, linking each of them as keys in `db` connected to a [Model](http://docs.sequelizejs.com/en/latest/api/model/)?

The [Model](http://docs.sequelizejs.com/en/latest/api/model/) class doesn't have any `associate` property in it, so I'm not quite sure what you're doing with `if ('associate' in db[modelName])`.
Am I missing something? I've only looked at it for 5 - 10 minutes, so it may be some conditional property I overlooked...

**Comment by sharminelow** [comment]
The associate was a self-defined attribute/method. To keep the initiation of each model separate and modular, I wanted the associations to be written inside each model file instead of putting them inside another Association.js or within Storage.js. And the associations methods (http://docs.sequelizejs.com/en/latest/api/associations/)  will be declared inside the model file to make the code cohesive. However, sequelize doesn't have a method of invoking those methods inside each model file. They will have to be written inside Storage.js if I didn't have that portion of code to call those methods. 

I had that portion of code removed for the latest commit, and I am closing this pull request. I suppose I can do more stuff before the sprint ends.

**Comment by yj-soh** [comment]
I thought about it yesterday, but how about declaring two objects under each model class?
So you would have `User.fields` (the current `User`) and `User.associations`.
You would first pass in `User.fields` using `sequelize.import`, then use `User.associations` to determine what associations are necessary after creating all of the models.
This allows you to keep everything within each model class + the ability to create every model before linking them together.

Something like this:

```
models.forEach(function(model) {
  model.associations.hasMany.forEach(function(linkedModel) {
    db[model].hasMany(db[linkedModel]);
});
```

**Comment by sharminelow** [comment]
I am using the method, `sequelize.define('User', args)`. I don't think I can nest them inside `User.fields` based on the docs. It's not a node.js object. Its a sequelize model object. For self-defined functions, attributes, I have to nest them inside `classMethods`.

**Comment by yj-soh** [comment]
Ah, I see, it won't work since it takes in a path-

...Or so I thought before looking it up further.
There's an alternate signature where it takes in a callback.
Try changing the import part to use that signature instead.

```
module.exports.getModel = function(sequelize, DataTypes){ ... };
module.exports.getAssociations = ...;
```

```
models.forEach(function(model) {
  db[model] = sequelize.import(model, require('model-path/' + model).getModel); // (not quite sure what the first arg for import is)
  logger.info(model + ' model imported');
});
models.forEach(function(model) {
  var associations = require('model-path/' + model).getAssociations;
  // do stuff
});
```

I think this will allow for what you had in mind?

**Comment by sharminelow** [comment]
Wow, that seems right. I will try that in future commits when we have more than just one model.

**Pull request closed without merging**